oeps/architectural-decisions/oep-0052/decisions/002-schema-specification.rst

```diff
@@ -0,0 +1,29 @@
+Event Bus Messages will be Defined in AVRO Specification Language
+=================================================================
+
+Context
+-------
+
+* Currently, Pulsar does not support `JSON Schema <https://json-schema.org/specification.html>`__, it only supports AVRO
+  for schema definition. Pulsar does support serializing AVRO to JSON (confusing using a ``JSONSchema`` class).
+
+* Industry best practices seem to suggest using a binary encoding of messages.
+
+* Industry best practice also suggests using AVRO over JSONSchema for better expression around schema evolution.
+
+* Even if we use a JSON encoding for the message we'll probably want to enable compression on the wire to reduce data
+  size.  This would mean whether or not we use JSON encoding is moot because the data on the wire will still be binary
+  encoded.
+
+Decision
+--------
+
+* We will use the AVRO specification to define the schema of messages that are sent over the event bus.
+
+* We will also use the binary serialization of messages that are transmitted over the event bus.
```

Comment:
Is it possible to switch between binary/JSON for debugging purposes and would the header let you know how to proceed?

From the [Avro docs on Encodings](https://avro.apache.org/docs/current/spec.html#Encodings): 
> Avro specifies two serialization encodings: binary and JSON. Most applications will use the binary encoding, as it is smaller and faster. But, for debugging and web-based applications, the JSON encoding may sometimes be appropriate.

Reply:
AvroAttrsBridge does not currently support that. Kafka might have a way for us to do this, but I don't know about it.